### PR TITLE
Add unregister method to NetDispatcher

### PR DESCRIPTION
**Pull request by kcudnik**

### CODE REVIEW

File: common/netdispatcher.h [hunk 1]
```diff
@@ -30,13 +33,32 @@ class NetDispatcher {
...
+
 private:
+
     NetDispatcher() = default;
 
+    NetDispatcher(const NetDispatcher&) = delete;
+
+    NetMsg* getCallback(int nlmsg_type);
+
+private:
```
Comment (qiluo-msft) [inline common/netdispatcher.h]: Not needed. Is it on purpose?

Comment (kcudnik) [reply common/netdispatcher.h]: yes, on purpose, look that its under mutex, so all operations on m_map are synchronized, also get must be protected, so hence t his function, i could do it lika { MUTEX; callbacl = m_map.get(); } but having function is much cleaner

File: common/netdispatcher.cpp [hunk 2]
```diff
@@ -4,34 +4,64 @@
 
 using namespace swss;
 
+#define MUTEX std::lock_guard<std::mutex> _lock(m_mutex);
```
Comment (qiluo-msft) [inline common/netdispatcher.cpp]: >mutex [](start = 35, length = 5)

Use spinlock to improve runtime performance, maybe atomic_flag?

Comment (kcudnik) [reply common/netdispatcher.cpp]: this is not performance critical ?

**Pull request merged**